src/plugins/data/public/search/search_service.ts

```diff
@@ -115,7 +117,9 @@ export class SearchService implements Plugin<ISearchSetup, ISearchStart> {
 
     const searchSourceDependencies: SearchSourceDependencies = {
       getConfig: uiSettings.get.bind(uiSettings),
-      search,
+      search: (request: IEsSearchRequest, options: ISearchOptions) => {
```

Comment:
@lukeelmers this change will be a problem for https://github.com/elastic/kibana/pull/78645, as the previous type was required to support EQL queries.